src/redux/books/books.js

```diff
@@ -0,0 +1,34 @@
+// Actions
+const ADD = 'bookstore/books/add';
+const DEL = 'bookstore/books/del';
+
+// Empty array of books
+const initialState = [];
+
+// Reducer
+export default function booksReducer(state = initialState, action = {}) {
+  switch (action.type) {
+    case ADD:
+      return [...state, {
+        ...action.payload,
+      }];
+    case DEL:
+      return state.filter((book) => book.id !== action.id);
+    default: return state;
+  }
+}
+
+// Action Creators
+export function add(book) {
+  return {
+    type: ADD,
+    payload: { book },
+  };
+}
+
+export function del(id) {
+  return {
+    type: DEL,
+    payload: { id },
+  };
```

Comment:
- [ ] It is not necessary to wrap the `id` Into the object, you can easily pass the `id` on `payload` which will help you to easily use it in the reducer while filtering.

  _**Suggestions:**_ ` return { type: DEL, payload: id }`